### PR TITLE
#1424: Defend against missing 'datasets' key in datasets/list response.

### LINKED ISSUE #1424: BigQuery client.list_datasets() fails when the project has no datasets
Found while doing clean-up of a project after reproducing #1419

Reproduce:

```python
In [11]: client.list_datasets()
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-11-fed4650eba20> in <module>()
----> 1 client.list_datasets()

gcloud-python/gcloud/bigquery/client.pyc in list_datasets(self, include_all, max_results, page_token)
     89                                            query_params=params)
     90         datasets = [Dataset.from_api_repr(resource, self)
---> 91                     for resource in resp['datasets']]
     92         return datasets, resp.get('nextPageToken')
     93 

KeyError: 'datasets'``

### PR DESCRIPTION
**Pull request by tseaver**
Fixes #1424.

### CODE REVIEW

File: gcloud/bigquery/test_client.py [hunk 1]
```diff
@@ -74,27 +74,11 @@ def test_list_datasets_defaults(self):
         self.assertEqual(req['method'], 'GET')
         self.assertEqual(req['path'], '/%s' % PATH)
 
-    def test_list_datasets_explicit(self):
```
Comment (dhermes) [inline gcloud/bigquery/test_client.py]: It seems you deleted the old test when you probably just wanted to add the new one.

**Comment by dhermes** [comment]
Can you also update the `system_test` as was done in #1421?

File: gcloud/bigquery/test_client.py (hunk 1 above)
Comment (tseaver) [inline gcloud/bigquery/test_client.py]: I saw that there was no need for a separate test:  the explicit one was redundant with the defaults one for testing the unmarshalling of the `Dataset` objects, so I just updated to return a set with no `datasets` key at all.

**Comment by tseaver** [comment]
> Can you also update the system_test as was done in #1421?

Tricky:  Datasets are top-level objects, so it isn't guaranteed that there won't be any already existing for a given project.

**Comment by dhermes** [comment]
Touche. Fine as is. I should have thought more than just the one tiny thought.

File: gcloud/bigquery/test_client.py (hunk 1 above)
Comment (dhermes) [inline gcloud/bigquery/test_client.py]: OK. SGTM.

**Comment by dhermes** [comment]
LGTM

**Pull request merged**